'platform' is not exported by node-resolve:empty.js

Hi,
When setting up a simple browser based project with snowpack (probably rollup as well). Importing contentful (`import {createClient} from 'contentful'`) generates the following error:

```
[snowpack] ! installing dependencies...
[snowpack] Failed to load node_modules/.pnpm/contentful-sdk-core@6.7.0/node_modules/contentful-sdk-core/dist/index.es-modules.js
  'platform' is not exported by node-resolve:empty.js, imported by node_modules/.pnpm/contentful-sdk-core@6.7.0/node_modules/contentful-sdk-core/dist/index.es-modules.js
[snowpack] Install failed.
```
Main tools used: Svelte, typescript, Snowpack


I have solved this simply by editing the file in question as suggested here:
https://github.com/snowpackjs/snowpack/discussions/1387#discussioncomment-117946
```
# file: node_modules/.pnpm/contentful-sdk-core@6.7.0/node_modules/contentful-sdk-core/dist/index.es-modules.js
- import { platform, release } from 'os';
+ import os from 'os';
```

But it would of course be nice if this was fixed by contentful!

Steps to reproduce
```
npx create-snowpack-app myproject --template @snowpack/app-template-svelte-typescript
cd myproject
npm i contentful
# edit src/App.svelte and add import 'contentful' inside <script> tag
npm start
```

Same issue happens with **SvelteKit** which uses **Vite**:

```
sveltekit % npm run dev --

> ~TODO~@0.0.1 dev /Users/***/dev/sveltekit
> svelte-kit dev


  SvelteKit v1.0.0-next.109

  local:   http://localhost:5000
  network: not exposed

  Use --host to expose server to other devices on this network


10:12:55 [vite] new dependencies found: contentful, updating...
 > node_modules/contentful-sdk-core/dist/index.es-modules.js:3:9: error: No matching export in "browser-external:os" for import "platform"
    3 │ import { platform, release } from 'os';
      ╵          ~~~~~~~~

 > node_modules/contentful-sdk-core/dist/index.es-modules.js:3:19: error: No matching export in "browser-external:os" for import "release"
    3 │ import { platform, release } from 'os';
      ╵                    ~~~~~~~

10:12:55 [vite] error while updating dependencies:
Error: Build failed with 2 errors:
node_modules/contentful-sdk-core/dist/index.es-modules.js:3:9: error: No matching export in "browser-external:os" for import "platform"
node_modules/contentful-sdk-core/dist/index.es-modules.js:3:19: error: No matching export in "browser-external:os" for import "release"
    at failureErrorWithLog (/Users/angel.meraz/dev/sveltekit/node_modules/esbuild/lib/main.js:1443:15)
    at /Users/angel.meraz/dev/sveltekit/node_modules/esbuild/lib/main.js:1125:28
    at runOnEndCallbacks (/Users/angel.meraz/dev/sveltekit/node_modules/esbuild/lib/main.js:915:63)
    at buildResponseToResult (/Users/angel.meraz/dev/sveltekit/node_modules/esbuild/lib/main.js:1123:7)
    at /Users/angel.meraz/dev/sveltekit/node_modules/esbuild/lib/main.js:1230:14
    at /Users/angel.meraz/dev/sveltekit/node_modules/esbuild/lib/main.js:606:9
    at handleIncomingPacket (/Users/angel.meraz/dev/sveltekit/node_modules/esbuild/lib/main.js:703:9)
    at Socket.readFromStdout (/Users/angel.meraz/dev/sveltekit/node_modules/esbuild/lib/main.js:573:7)
    at Socket.emit (events.js:315:20)
    at Socket.EventEmitter.emit (domain.js:467:12)
```

However it does work by updating `contentful-sdk-core/dist/index.es-modules.js` as suggested by @jesperp

@luizfonseca I gave this a try and got the following error:
```sveltekit-contentful-app % npm run dev --

> sveltekit-contentful-app@0.0.1 dev /Users/angel.meraz/dev/sveltekit-contentful-app
> svelte-kit dev


  SvelteKit v1.0.0-next.109

  local:   http://localhost:3000
  network: not exposed

  Use --host to expose server to other devices on this network


10:35:23 [vite] hmr update /src/routes/index.svelte
10:35:28 [vite] new dependencies found: contentful, updating...
 > node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:14: error: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "contentful-sdk-core". The package may have incorrect main/module/exports specified in its package.json.
    32580 │         throw new Error(`Failed to resolve entry for package "${id}". ` +
          ╵               ^
    at resolvePackageEntry (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:15)
    at tryNodeResolve (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32399:11)
    at Context.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32282:28)
    at Object.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:44228:55)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:69926:27
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64628:34
    at async callback (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:809:28)
    at async handleRequest (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:641:30)

   node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:18: note: This error came from the "onResolve" callback registered here
    64615 │             build.onResolve({ filter: /^[\w@][^:]/ }, async ({ path: id, importer, kind, resolveDir }) => {
          ╵                   ~~~~~~~~~
    at setup (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:19)
    at handlePlugins (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:733:23)
    at Object.buildOrServe (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1018:7)
    at /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1742:17
    at new Promise (<anonymous>)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1741:14)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1617:51)
    at optimizeDeps (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64842:34)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

   node_modules/contentful/dist/es-modules/contentful.js:10:33: note: The plugin "vite:dep-pre-bundle" was triggered by this import
       10 │ var _contentfulSdkCore = require("contentful-sdk-core");
          ╵                                  ~~~~~~~~~~~~~~~~~~~~~

 > node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:14: error: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "contentful-sdk-core". The package may have incorrect main/module/exports specified in its package.json.
    32580 │         throw new Error(`Failed to resolve entry for package "${id}". ` +
          ╵               ^
    at resolvePackageEntry (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:15)
    at tryNodeResolve (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32399:11)
    at Context.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32282:28)
    at Object.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:44228:55)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:69926:27
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64628:34
    at async callback (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:809:28)
    at async handleRequest (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:641:30)

   node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:18: note: This error came from the "onResolve" callback registered here
    64615 │             build.onResolve({ filter: /^[\w@][^:]/ }, async ({ path: id, importer, kind, resolveDir }) => {
          ╵                   ~~~~~~~~~
    at setup (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:19)
    at handlePlugins (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:733:23)
    at Object.buildOrServe (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1018:7)
    at /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1742:17
    at new Promise (<anonymous>)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1741:14)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1617:51)
    at optimizeDeps (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64842:34)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

   node_modules/contentful/dist/es-modules/create-contentful-api.js:8:33: note: The plugin "vite:dep-pre-bundle" was triggered by this import
        8 │ var _contentfulSdkCore = require("contentful-sdk-core");
          ╵                                  ~~~~~~~~~~~~~~~~~~~~~

 > node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:14: error: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "contentful-sdk-core". The package may have incorrect main/module/exports specified in its package.json.
    32580 │         throw new Error(`Failed to resolve entry for package "${id}". ` +
          ╵               ^
    at resolvePackageEntry (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:15)
    at tryNodeResolve (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32399:11)
    at Context.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32282:28)
    at Object.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:44228:55)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:69926:27
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64628:34
    at async callback (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:809:28)
    at async handleRequest (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:641:30)

   node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:18: note: This error came from the "onResolve" callback registered here
    64615 │             build.onResolve({ filter: /^[\w@][^:]/ }, async ({ path: id, importer, kind, resolveDir }) => {
          ╵                   ~~~~~~~~~
    at setup (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:19)
    at handlePlugins (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:733:23)
    at Object.buildOrServe (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1018:7)
    at /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1742:17
    at new Promise (<anonymous>)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1741:14)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1617:51)
    at optimizeDeps (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64842:34)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

   node_modules/contentful/dist/es-modules/entities/space.js:8:33: note: The plugin "vite:dep-pre-bundle" was triggered by this import
        8 │ var _contentfulSdkCore = require("contentful-sdk-core");
          ╵                                  ~~~~~~~~~~~~~~~~~~~~~

 > node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:14: error: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "contentful-sdk-core". The package may have incorrect main/module/exports specified in its package.json.
    32580 │         throw new Error(`Failed to resolve entry for package "${id}". ` +
          ╵               ^
    at resolvePackageEntry (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:15)
    at tryNodeResolve (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32399:11)
    at Context.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32282:28)
    at Object.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:44228:55)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:69926:27
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64628:34
    at async callback (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:809:28)
    at async handleRequest (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:641:30)

   node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:18: note: This error came from the "onResolve" callback registered here
    64615 │             build.onResolve({ filter: /^[\w@][^:]/ }, async ({ path: id, importer, kind, resolveDir }) => {
          ╵                   ~~~~~~~~~
    at setup (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:19)
    at handlePlugins (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:733:23)
    at Object.buildOrServe (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1018:7)
    at /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1742:17
    at new Promise (<anonymous>)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1741:14)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1617:51)
    at optimizeDeps (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64842:34)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

   node_modules/contentful/dist/es-modules/paged-sync.js:8:33: note: The plugin "vite:dep-pre-bundle" was triggered by this import
        8 │ var _contentfulSdkCore = require("contentful-sdk-core");
          ╵                                  ~~~~~~~~~~~~~~~~~~~~~

 > node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:14: error: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "contentful-sdk-core". The package may have incorrect main/module/exports specified in its package.json.
    32580 │         throw new Error(`Failed to resolve entry for package "${id}". ` +
          ╵               ^
    at resolvePackageEntry (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:15)
    at tryNodeResolve (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32399:11)
    at Context.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32282:28)
    at Object.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:44228:55)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:69926:27
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64628:34
    at async callback (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:809:28)
    at async handleRequest (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:641:30)

   node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:18: note: This error came from the "onResolve" callback registered here
    64615 │             build.onResolve({ filter: /^[\w@][^:]/ }, async ({ path: id, importer, kind, resolveDir }) => {
          ╵                   ~~~~~~~~~
    at setup (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:19)
    at handlePlugins (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:733:23)
    at Object.buildOrServe (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1018:7)
    at /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1742:17
    at new Promise (<anonymous>)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1741:14)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1617:51)
    at optimizeDeps (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64842:34)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

   node_modules/contentful/dist/es-modules/entities/asset-key.js:10:33: note: The plugin "vite:dep-pre-bundle" was triggered by this import
       10 │ var _contentfulSdkCore = require("contentful-sdk-core");
          ╵                                  ~~~~~~~~~~~~~~~~~~~~~

 > node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:14: error: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "contentful-sdk-core". The package may have incorrect main/module/exports specified in its package.json.
    32580 │         throw new Error(`Failed to resolve entry for package "${id}". ` +
          ╵               ^
    at resolvePackageEntry (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:15)
    at tryNodeResolve (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32399:11)
    at Context.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32282:28)
    at Object.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:44228:55)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:69926:27
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64628:34
    at async callback (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:809:28)
    at async handleRequest (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:641:30)

   node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:18: note: This error came from the "onResolve" callback registered here
    64615 │             build.onResolve({ filter: /^[\w@][^:]/ }, async ({ path: id, importer, kind, resolveDir }) => {
          ╵                   ~~~~~~~~~
    at setup (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:19)
    at handlePlugins (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:733:23)
    at Object.buildOrServe (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1018:7)
    at /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1742:17
    at new Promise (<anonymous>)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1741:14)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1617:51)
    at optimizeDeps (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64842:34)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

   node_modules/contentful/dist/es-modules/entities/asset.js:11:33: note: The plugin "vite:dep-pre-bundle" was triggered by this import
       11 │ var _contentfulSdkCore = require("contentful-sdk-core");
          ╵                                  ~~~~~~~~~~~~~~~~~~~~~

 > node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:14: error: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "contentful-sdk-core". The package may have incorrect main/module/exports specified in its package.json.
    32580 │         throw new Error(`Failed to resolve entry for package "${id}". ` +
          ╵               ^
    at resolvePackageEntry (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:15)
    at tryNodeResolve (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32399:11)
    at Context.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32282:28)
    at Object.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:44228:55)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:69926:27
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64628:34
    at async callback (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:809:28)
    at async handleRequest (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:641:30)

   node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:18: note: This error came from the "onResolve" callback registered here
    64615 │             build.onResolve({ filter: /^[\w@][^:]/ }, async ({ path: id, importer, kind, resolveDir }) => {
          ╵                   ~~~~~~~~~
    at setup (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:19)
    at handlePlugins (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:733:23)
    at Object.buildOrServe (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1018:7)
    at /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1742:17
    at new Promise (<anonymous>)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1741:14)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1617:51)
    at optimizeDeps (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64842:34)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

   node_modules/contentful/dist/es-modules/entities/locale.js:11:33: note: The plugin "vite:dep-pre-bundle" was triggered by this import
       11 │ var _contentfulSdkCore = require("contentful-sdk-core");
          ╵                                  ~~~~~~~~~~~~~~~~~~~~~

 > node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:14: error: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "contentful-sdk-core". The package may have incorrect main/module/exports specified in its package.json.
    32580 │         throw new Error(`Failed to resolve entry for package "${id}". ` +
          ╵               ^
    at resolvePackageEntry (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:15)
    at tryNodeResolve (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32399:11)
    at Context.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32282:28)
    at Object.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:44228:55)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:69926:27
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64628:34
    at async callback (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:809:28)
    at async handleRequest (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:641:30)

   node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:18: note: This error came from the "onResolve" callback registered here
    64615 │             build.onResolve({ filter: /^[\w@][^:]/ }, async ({ path: id, importer, kind, resolveDir }) => {
          ╵                   ~~~~~~~~~
    at setup (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:19)
    at handlePlugins (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:733:23)
    at Object.buildOrServe (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1018:7)
    at /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1742:17
    at new Promise (<anonymous>)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1741:14)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1617:51)
    at optimizeDeps (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64842:34)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

   node_modules/contentful/dist/es-modules/entities/content-type.js:11:33: note: The plugin "vite:dep-pre-bundle" was triggered by this import
       11 │ var _contentfulSdkCore = require("contentful-sdk-core");
          ╵                                  ~~~~~~~~~~~~~~~~~~~~~

 > node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:14: error: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "contentful-sdk-core". The package may have incorrect main/module/exports specified in its package.json.
    32580 │         throw new Error(`Failed to resolve entry for package "${id}". ` +
          ╵               ^
    at resolvePackageEntry (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:15)
    at tryNodeResolve (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32399:11)
    at Context.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32282:28)
    at Object.resolveId (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:44228:55)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:69926:27
    at async /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64628:34
    at async callback (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:809:28)
    at async handleRequest (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:641:30)

   node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:18: note: This error came from the "onResolve" callback registered here
    64615 │             build.onResolve({ filter: /^[\w@][^:]/ }, async ({ path: id, importer, kind, resolveDir }) => {
          ╵                   ~~~~~~~~~
    at setup (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64615:19)
    at handlePlugins (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:733:23)
    at Object.buildOrServe (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1018:7)
    at /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1742:17
    at new Promise (<anonymous>)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1741:14)
    at Object.build (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1617:51)
    at optimizeDeps (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:64842:34)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

   node_modules/contentful/dist/es-modules/entities/entry.js:11:33: note: The plugin "vite:dep-pre-bundle" was triggered by this import
       11 │ var _contentfulSdkCore = require("contentful-sdk-core");
          ╵                                  ~~~~~~~~~~~~~~~~~~~~~

10:35:29 [vite] error while updating dependencies:
Error: Build failed with 9 errors:
node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:14: error: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "contentful-sdk-core". The package may have incorrect main/module/exports specified in its package.json.
node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:14: error: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "contentful-sdk-core". The package may have incorrect main/module/exports specified in its package.json.
node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:14: error: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "contentful-sdk-core". The package may have incorrect main/module/exports specified in its package.json.
node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:14: error: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "contentful-sdk-core". The package may have incorrect main/module/exports specified in its package.json.
node_modules/vite/dist/node/chunks/dep-6b5f3ba8.js:32580:14: error: [plugin: vite:dep-pre-bundle] Failed to resolve entry for package "contentful-sdk-core". The package may have incorrect main/module/exports specified in its package.json.
...
    at failureErrorWithLog (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1443:15)
    at /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1125:28
    at runOnEndCallbacks (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:915:63)
    at buildResponseToResult (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1123:7)
    at /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1230:14
    at /Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:606:9
    at handleIncomingPacket (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:703:9)
    at Socket.readFromStdout (/Users/angel.meraz/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:573:7)
    at Socket.emit (events.js:315:20)
    at Socket.EventEmitter.emit (domain.js:467:12)
```

package.json
```
{
  "name": "sveltekit-contentful-app",
  "version": "0.0.1",
  "scripts": {
    "dev": "svelte-kit dev",
    "build": "svelte-kit build",
    "preview": "svelte-kit preview",
    "lint": "prettier --check --plugin-search-dir=. . && eslint --ignore-path .gitignore .",
    "format": "prettier --write --plugin-search-dir=. ."
  },
  "devDependencies": {
    "@sveltejs/kit": "next",
    "@typescript-eslint/eslint-plugin": "^4.19.0",
    "@typescript-eslint/parser": "^4.19.0",
    "contentful-sdk-core": "git://github.com/contentful/contentful-sdk-core.git#fix/os-error",
    "eslint": "^7.22.0",
    "eslint-config-prettier": "^8.1.0",
    "eslint-plugin-svelte3": "^3.2.0",
    "prettier": "~2.2.1",
    "prettier-plugin-svelte": "^2.2.0",
    "svelte": "^3.34.0",
    "svelte-preprocess": "^4.0.0",
    "tslib": "^2.0.0",
    "typescript": "^4.0.0",
    "vite": "^2.3.4"
  },
  "type": "module",
  "dependencies": {
    "contentful": "^8.3.6",
    "dotenv": "^10.0.0",
    "path": "^0.12.7"
  }
}

```

My bad there. Here it goes again, still having the same issue as before.

```> sveltekit-contentful-app@0.0.1 dev /Users/****/dev/sveltekit-contentful-app
> svelte-kit dev


  SvelteKit v1.0.0-next.111

  local:   http://localhost:3000
  network: not exposed

  Use --host to expose server to other devices on this network


09:08:25 [vite] new dependencies found: contentful, updating...
 > node_modules/contentful/node_modules/contentful-sdk-core/dist/index.es-modules.js:3:9: error: No matching export in "browser-external:os" for import "platform"
    3 │ import { platform, release } from 'os';
      ╵          ~~~~~~~~

 > node_modules/contentful/node_modules/contentful-sdk-core/dist/index.es-modules.js:3:19: error: No matching export in "browser-external:os" for import "release"
    3 │ import { platform, release } from 'os';
      ╵                    ~~~~~~~

09:08:26 [vite] error while updating dependencies:
Error: Build failed with 2 errors:
node_modules/contentful/node_modules/contentful-sdk-core/dist/index.es-modules.js:3:9: error: No matching export in "browser-external:os" for import "platform"
node_modules/contentful/node_modules/contentful-sdk-core/dist/index.es-modules.js:3:19: error: No matching export in "browser-external:os" for import "release"
    at failureErrorWithLog (/Users/****/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1443:15)
    at /Users/****/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1125:28
    at runOnEndCallbacks (/Users/****/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:915:63)
    at buildResponseToResult (/Users/****/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1123:7)
    at /Users/****/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:1230:14
    at /Users/****/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:606:9
    at handleIncomingPacket (/Users/****/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:703:9)
    at Socket.readFromStdout (/Users/****/dev/sveltekit-contentful-app/node_modules/esbuild/lib/main.js:573:7)
    at Socket.emit (events.js:315:20)
    at Socket.EventEmitter.emit (domain.js:467:12)
```

package.json

```
{
  "name": "sveltekit-contentful-app",
  "version": "0.0.1",
  "scripts": {
    "dev": "svelte-kit dev",
    "build": "svelte-kit build",
    "preview": "svelte-kit preview",
    "lint": "prettier --check --plugin-search-dir=. . && eslint --ignore-path .gitignore .",
    "format": "prettier --write --plugin-search-dir=. ."
  },
  "devDependencies": {
    "@sveltejs/kit": "next",
    "@typescript-eslint/eslint-plugin": "^4.19.0",
    "@typescript-eslint/parser": "^4.19.0",
    "eslint": "^7.22.0",
    "eslint-config-prettier": "^8.1.0",
    "eslint-plugin-svelte3": "^3.2.0",
    "prettier": "~2.2.1",
    "prettier-plugin-svelte": "^2.2.0",
    "svelte": "^3.34.0",
    "svelte-preprocess": "^4.0.0",
    "tslib": "^2.0.0",
    "typescript": "^4.0.0",
    "vite": "^2.3.4"
  },
  "type": "module",
  "dependencies": {
    "contentful": "^8.3.6",
    "contentful-sdk-core": "git://github.com/contentful/contentful-sdk-core.git#fix/os-error",
    "dotenv": "^10.0.0",
    "path": "^0.12.7"
  }
}
```

--- a/src/get-user-agent.ts
+++ b/src/get-user-agent.ts
@@ -1,4 +1,4 @@
-import { platform, release } from 'os'
+import os from 'os'
 
 import { isNode, getNodeVersion, isReactNative, getWindow } from './utils'
 
@@ -12,29 +12,28 @@ function getBrowserOS(): string | null {
   const macosPlatforms = ['Macintosh', 'MacIntel', 'MacPPC', 'Mac68K']
   const windowsPlatforms = ['Win32', 'Win64', 'Windows', 'WinCE']
   const iosPlatforms = ['iPhone', 'iPad', 'iPod']
-  let os = null
 
   if (macosPlatforms.indexOf(platform) !== -1) {
-    os = 'macOS'
+    return 'macOS'
   } else if (iosPlatforms.indexOf(platform) !== -1) {
-    os = 'iOS'
+    return 'iOS'
   } else if (windowsPlatforms.indexOf(platform) !== -1) {
-    os = 'Windows'
+    return 'Windows'
   } else if (/Android/.test(userAgent)) {
-    os = 'Android'
+    return 'Android'
   } else if (/Linux/.test(platform)) {
-    os = 'Linux'
+    return 'Linux'
   }
 
-  return os
+  return null
 }
 
-type OsMap = Record<string,  'Android' | 'Linux' | 'Windows' | 'macOS'>
+type PlatformMap = Record<string,  'Android' | 'Linux' | 'Windows' | 'macOS'>
 
 function getNodeOS(): string | null {
-  const os = platform() || 'linux'
-  const version = release() || '0.0.0'
-  const osMap: OsMap = {
+  const platform = os.platform() || 'linux'
+  const version = os.release() || '0.0.0'
+  const platformMap: PlatformMap = {
     android: 'Android',
     aix: 'Linux',
     darwin: 'macOS',
@@ -44,8 +43,8 @@ function getNodeOS(): string | null {
     sunos: 'Linux',
     win32: 'Windows',
   }
-  if (os in osMap) {
-    return `${osMap[os] || 'Linux'}/${version}`
+  if (platform in platformMap) {
+    return `${platformMap[platform] || 'Linux'}/${version}`
   }
   return null
 }
@@ -72,24 +71,24 @@ export default function getUserAgentHeader(
 
   headerParts.push(`sdk ${sdk}`)
 
-  let os = null
+  let platform = null
   try {
     if (isReactNative()) {
-      os = getBrowserOS()
+      platform = getBrowserOS()
       headerParts.push('platform ReactNative')
     } else if (isNode()) {
-      os = getNodeOS()
+      platform = getNodeOS()
       headerParts.push(`platform node.js/${getNodeVersion()}`)
     } else {
-      os = getBrowserOS()
+      platform = getBrowserOS()
       headerParts.push('platform browser')
     }
   } catch (e) {
-    os = null
+    platform = null
   }
 
-  if (os) {
-    headerParts.push(`os ${os}`)
+  if (platform) {
+    headerParts.push(`os ${platform}`)
   }
 
   return `${headerParts.filter((item) => item !== '').join('; ')};`

